DeviceCode/Targets/Native/STM32F4/DeviceCode/STM32F4_CRC/CRC_functions.cpp

```diff
@@ -0,0 +1,94 @@
+//////////////////////////////////////////////////////////////////////////////////////////
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Copyright (c) Microsoft Open Technologies. All rights reserved.
+//  Implementation for CMSIS: Copyright (c) Eclo Solutions Lda
+//
+//  *** Interface for Cortex-M CRC Calculation Unit ***
+//
+//////////////////////////////////////////////////////////////////////////////////////////
+
+#include <tinyhal.h>
+
+#define CRC_BASE            (AHB1PERIPH_BASE + 0x3000U)
+#define CRC                 ((CRC_TypeDef *) CRC_BASE)
+
+typedef struct
+{
+  CRC_TypeDef                 *Instance;  /*!< Register base address   */
+}CRC_HandleTypeDef;
+
+
+/*
+    Interface to Cortex-M CRC calculation unit that implements the equivalent to the software implementation at Support\CRC project.
+    CRC-32 (Ethernet) polynomial: 0x4C11DB7.
+    buffer: pointer to the region block to be CRCed
+    size: lenght of buffer to compute CRC
+    crc: previous CRC value to start CRC computing
+*/
+
+/* NOTES: 
+    - should be good for F1, L1, F2 and F4 units series, other have diferent configurations and polynomial coefficients
+    - can only handle buffer sizes multiple of 4 otherwise the calculation is incorrect
+    - can not handle initial CRC values, only 0 is admited
+    - all the above can be fixed with more code so feel free to jump in ;)     
+*/
+
+
+UINT32 SUPPORT_ComputeCRC(const void* buffer, int size, UINT32 initCrc)

```

Comment:
Do you have any performance measurements on the hardware based solution vs. software? The table based software approach is pretty small in code size, big on data size for the table and reasonably fast, so I'm curios to see how much better the hardware form is.